### PR TITLE
[BB-3583] Unit Test: Test Registration Page Frontend and Coverage

### PR DESCRIPTION
**Pull request by spokerman12**
**Description**

We're fulfilling the following user story:

> "As a OCIM developer, I want to unit test the registration page so that making changes are faster"

**Supporting information**

- [Jira ticket](https://tasks.opencraft.com/browse/BB-3583)

**Testing instructions**

1. Set up a local OCIM environment using this branch
2. Execute `npm run coverage`
3. Open `frontend/coverage/lcov-report/index.html` in a browser.
4. Check package.json, see if the settings are acceptable.
5. Execute `npm test registration` to run the new unit tests

**Snapshots:**

None

**Further considerations:**

- We're only doing unit tests for the components in `frontend/registration`
- We should contemplate how much granularity we want for unit tests, e.g.: should we unit test reducers or just their results?

Reviewer:
[ ] @farhaanbukhsh

### CODE REVIEW

File: frontend/src/registration/components/RegistrationPage/RegistrationPage.spec.tsx [hunk 1]
```diff
@@ -1,8 +1,48 @@
...
+        currentStep={0}
+      />,
+    );
+    const mountedComponent = tree.find('RegistrationPage')
+
+    expect(mountedComponent.find('h1').at(0).text()).toBe('test-title');
+    expect(mountedComponent.find('h1').at(1).text()).toBe('test-subtitleBig');
+    expect(mountedComponent.find('h2').text()).toBe('test-subtitle');
+    expect(mountedComponent.find('StepBar').at(0).prop('currentStep')).toBe(0);
+    expect(mountedComponent.find('div.registration-page-content').text()).toBe('test-child')
```
Comment (farhaanbukhsh) [inline frontend/src/registration/components/RegistrationPage/RegistrationPage.spec.tsx:31]: :+1:

**Review by farhaanbukhsh — COMMENTED** [review]
I have left a few comments but above that, it looks good.

File: frontend/src/utils/testing.tsx [hunk 2]
```diff
@@ -6,8 +6,12 @@ import { MemoryRouter } from 'react-router';
 import * as renderer from 'react-test-renderer';
 import { applyMiddleware, createStore } from 'redux';
 import thunk from 'redux-thunk';
+import { shallow, configure, render, mount } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
 import { createRootReducer } from '../global/reducers';
```
[the remark this answers is not in the document]
Comment (spokerman12) [reply frontend/src/utils/testing.tsx:12]: I'd agree. Initially I wanted to add some external links to this doc. I'll add a brief explanation for each method

File: frontend/src/registration/components/AccountSetupPage/AccountSetupPage.spec.tsx [hunk 3]
```diff
@@ -1,8 +1,90 @@
...
+      expect(mountedComponent.state).toStrictEqual({
+        fullName: 'Some Fullname',
+        username: 'someUsername',
+        email: 'some@email.com',
+        password: 'Somepassword1#',
+        passwordConfirm: 'Somepassword1#',
+        acceptTOS: true,
+        acceptDomainCondition: true,
+        subscribeToUpdates: true
+      })
```
[the remark this answers is not in the document]
Comment (spokerman12) [reply frontend/src/registration/components/AccountSetupPage/AccountSetupPage.spec.tsx:87]: Sure thing

File: frontend/src/registration/components/InstanceSetupPage/__snapshots__/InstanceSetupPage.spec.tsx.snap [hunk 4]
```diff
@@ -1,6 +1,6 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`renders without crashing 1`] = `
+exports[`RedirectToCorrect Step component renders without crashing 1`] = `
```
[the remark this answers is not in the document]
Comment (spokerman12) [reply frontend/src/registration/components/InstanceSetupPage/__snapshots__/InstanceSetupPage.spec.tsx.snap:3]: it's just the name of the test that's forcing this change.

File: frontend/src/registration/components/RedirectToCorrectStep/RedirectToCorrectStep.spec.tsx [hunk 5]
```diff
@@ -1,8 +1,68 @@
...
+            />
+        );
+        const mountedComponent = tree.find('RedirectToCorrectStep')
+        expect(
+            mountedComponent.prop('currentPageStep')
+        ).toBeGreaterThan(
+            mountedComponent.prop('currentRegistrationStep')
+        )
+        expect(mountedComponent.find('Redirect').prop('to')).toBe(REGISTRATION_STEPS[0]);
+    });
```
[the remark this answers is not in the document]
Comment (spokerman12) [reply frontend/src/registration/components/RedirectToCorrectStep/RedirectToCorrectStep.spec.tsx:35]: Sure!

**Review by farhaanbukhsh — APPROVED** [review]
👍 

- [x] I tested this: I read through the test cases and ran it locally, which works fine for me.
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~
- [ ] ~~Includes documentation~~
- [ ] ~~I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~~

**Pull request merged**